src/commands/build.ts

```diff
@@ -37,10 +37,14 @@ export default defineCommand({
 
     const kit = await loadKit(cwd)
 
-    const nitroPreset = ctx.args.prerender ? 'static' : ctx.args.preset || process.env.NITRO_PRESET || process.env.SERVER_PRESET
+    const resolvedNitroPreset
+      = ctx.args.preset || process.env.NITRO_PRESET || process.env.SERVER_PRESET
+    const nitroPreset = ctx.args.prerender ? 'static' : resolvedNitroPreset
```

Comment:
in fact this is still not quite right - there are several static presets, including `vercel-static`, `netlify-static`, and so on, and we probably need to ignore (that is, not log warnings) for any static preset that's passed

Reply:
It looks like nitro tries to infer the correct static preset based on the value of `preset` when `static: true` (https://github.com/nitrojs/nitro/pull/2860) it's undocumented though (https://github.com/nitrojs/nitro/issues/1524). I just removed the logging and only fall back to static preset if none is provided during prerender, is this what you had in mind?

Reply:
we should not set the static preset but instead pass `static: true`

you can test by setting (for example) `NETLIFY=1 nuxi build --prerender` - we want it in that case to use netlify-static preset.

Reply:
Okay I think/hope I finally understand, I removed the overriding of the preset variable and now log the resolved nitro preset instead.